vm.go

```diff
@@ -286,6 +286,7 @@ type vm struct {
 	pc           int
 	stack        valueStack
 	sp, sb, args int
+	tmpValues    valueStack
```

Comment:
I could see how using the valueStack type wouldn't be desired, we could also use `[]Value` here, I would just need to duplicate the corresponding [MemUsage](https://github.com/mongodb-forks/goja/blob/097882fe6827a5988343f5cc6bca6c0a5e0f247e/vm.go#L5594) method that I get for free by using valueStack.

Reply:
I think using `valueStack` makes this a little confusing. Could we just move the definition of the `(s valueStack) MemUsage` function into a new helper function that's used by both MemUsage functions?

Reply:
Ah yes I like the idea of a slice of Values explicitly here too.